Trusty: Support blocking PRs through reviews

puerco opened the pull request:
# Summary

Minder now has the capability to request changes in PRs when it finds something odd based on Trusty dependency data.

This PR  also introduces a new setting in the trusty ruletype configuration to disable blocking when malicious deps are detected (all power to the users, but why would you want that?!).

Fixes https://github.com/stacklok/minder/issues/3380
Part of https://github.com/stacklok/minder/issues/3379

## Change Type

***Mark the type of change your PR introduces:***

- [ ] Bug fix (resolves an issue without affecting existing features)
- [x] Feature (adds new functionality without breaking changes)
- [ ] Breaking change (may impact existing functionalities or require documentation updates)
- [ ] Documentation (updates or additions to documentation)
- [ ] Refactoring or test improvements (no bug fixes or new functionality)

# Testing

![image](https://github.com/stacklok/minder/assets/3935899/38b64510-3397-49f7-9d17-5ce5fc048d57)

# Review Checklist:

- [x] Reviewed my own code for quality and clarity.
- [ ] Added comments to complex or tricky code sections.
- [x] Updated any affected documentation.
- [ ] Included tests that validate the fix or feature.
- [ ] Checked that related changes are merged.

rdimitrov commented on internal/engine/eval/trusty/config.go:
```diff
@@ -49,15 +49,19 @@ type ecosystemConfig struct {
-		Action: pr_actions.ActionSummary,
+		Action: pr_actions.ActionReviewPr,
+		Block: map[string]bool{
+			"malicious": true,
```
Isn't it safer if we invert this and don't rely on initialising the bool with true? 

Also I think we can even delete this option in the Review action. I mean turning it on kind of overlaps with the other actions and if someone doesn't want Minder to block the PR, they can use some of them, right? 

I think this is how the vulnerability/OSV rule works like, right?

puerco replied:
Ah so the idea is that this option is _for_ the review action. In addition to malicious, we will have another entry to block (or not) when deprecated packages are found. This will let users control when they want to block or just get feedback from minder separately for those two specific dependency types.

puerco replied:
Also note that this is initializing missing map entries from the default settings [here](https://github.com/stacklok/minder/pull/3392/files#diff-4747ed96b4116a79071ea012528f62f247e7f143e6d8e7f276739816cb54d163R179-R181).

jhrozek replied:
I'm actually fine with this config granularity @rdimitrov . The way I read it the action tells you the "what" to do with the PR and the "block" tells you which of the attributes to take into account when deciding whether to block. And I also think all of them should be on by default.

rdimitrov replied:
Ah, thanks for flipping the setting! 😃 Yeah, my motivation was to keep vulncheck and trusty behaviour similar, but I agree on the functionality of this + we'll be refactoring that part so we might as well make it available for all PR related Minder actions not just Trusty.

rdimitrov reviewed (changes requested):
I don't have anything against, but I'll click the request changes in case someone decides to merge it before we resolve the comment 👍

jhrozek reviewed (dismissed):
A test with a malicious package went fine. As discussed with @puerco it would be nice to display a list of alternatives to the user also when flagging a malicous package, but this can be done in a separate PR.

puerco commented:
In the rush to write this yesterday I noticed that the configuration flags I added didn't match those in the PRD. I rewrote the logic to rename the options and to make them configurable per ecosystem. @rdimitrov you'll be glad to see that the logic of the blocking flags is now flipped :)
